### DISCUSSION TITLE
How do I send a sysex message from mysql?

### DISCUSSION THREAD

**Discussion by PeterKarsson** [djipco/webmidi#400] [General]
I wonder what I am doing wrong when I am trying to send a sysex message with data from mysql?

This work. It sends out the data.
```js
output_device.send([0xF0, 0x00, 0x01, 0x55, 0x12, 0x01, 0x63, 0x01, 0x00, 0xF7]);
```
If I get data from mysql it will not work. The data shows in the alert but is not sent out.
```js
$.ajax({
dataType: "json",
type: "POST",
async: false,
url: 'get_message.php',
cache: false,
success: function(data)
{
alert(data[0]);
output_device.send([data[0]]);
}
});
```

**Comment by djipco** [comment]
To help you, we would have to know exactly what is inside `data[0]`.

**Reply by PeterKarsson** [reply]
This is what inside data[0]

```
$array = [];
$result = mysqli_query($conn, "SELECT sysex_request FROM device");	
$row = mysqli_fetch_assoc($result);
$sysex_request = $row['sysex_request'];
$array[0] = $sysex_request;

// send result back	
echo json_encode($array);
```

This is whats inside the row in mysql.

`0xF0, 0x00, 0x01, 0x55, 0x12, 0x01, 0x63, 0x01, 0x00, 0xF7`

**Reply by djipco** [reply]
Can you simply do `console.log([data[0]])` and show us the full output?

**Reply by PeterKarsson** [reply]
I have tested it now and this is the console error I get when I do...

`output_device.send([data[0]]);`

data[0] = 0xF0, 0x00, 0x01, 0x55, 0x12, 0x01, 0x33, 0x33, 0x33, 0xF7

`webmidi.iife.js:4418 Uncaught TypeError: Failed to execute 'send' on 'MIDIOutput': Running status is not allowed at index 0 (0).
    at Output.send (webmidi.iife.js:4418:24)
    at Object.success ((index):445:17)
    at c (jquery.min.js:2:25304)
    at Object.fireWith [as resolveWith] (jquery.min.js:2:26053)
    at l (jquery.min.js:2:77782)
    at XMLHttpRequest.<anonymous> (jquery.min.js:2:80265)
    at Object.send (jquery.min.js:2:80625)
    at Function.ajax (jquery.min.js:2:76167)
    at download_from_device ((index):431:4)
    at HTMLButtonElement.onclick ((index):501:108)`

I will not get any console error when I do...

`output_device.send([0xF0, 0x00, 0x01, 0x55, 0x12, 0x01, 0x63, 0x01, 0x00, 0xF7]);`

**Reply by djipco** [reply]
As you still have not provided the output of `console.log([data[0]])`, it is again not clear what you are sending.  If `data[0]` contains `[0xF0, 0x00, 0x01, 0x55, 0x12, 0x01, 0x33, 0x33, 0x33, 0xF7]`, then `[data[0]]` actually contains an array of an array (which is not a valid syntax for the [`Output.send()`](https://webmidijs.org/api/classes/Output#send) method of WEBMIDI.js):

```js
Output.send([[0xF0, 0x00, 0x01, 0x55, 0x12, 0x01, 0x33, 0x33, 0x33, 0xF7]]); // this is not allowed
```

By the way, [running status](http://midi.teragonaudio.com/tech/midispec/run.htm) was a way to compress MIDI streams. It is not supported by the [`MIDIOutput.send()`](https://www.w3.org/TR/webmidi/#midioutput-interface) method of Web MIDI API.  This means that the first byte of each message must be a status byte and not the value `0`.

**Reply by PeterKarsson** [reply]
Do you mean this?

`Array(1)
0
: 
"0xF0, 0x00, 0x01, 0x55, 0x12, 0x01, 0x33, 0x33, 0x33, 0xF7"
length
: 
1
[[Prototype]]
: 
Array(0)`

**Reply by djipco** [reply]
Yes, this is exactly what I mean.

What you currently have is a 1-position array that contains a text string. This IS NOT what you need. What you need is a 10-position array that contains the actual bytes.

**Reply by PeterKarsson** [reply]
Ok I will test that and come back to you with the result.

**Reply by PeterKarsson** [reply]
I have tested it now and it works. This is the code I used.

`sysex_data_string = data[0];

const sysex_data_array = sysex_data_string.split(",");

output_device.send(sysex_data_array);

console.log([data[0]]);`
